waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -149,14 +150,31 @@ validateAuthUserEntityHasCorrectFieldsIfUsernameAndPasswordAuthIsUsed spec = cas
   Just auth ->
     if not $ Auth.isUsernameAndPasswordAuthEnabled auth
       then []
-      else
-        let userEntity = snd $ AS.resolveRef spec (Auth.userEntity auth)
-            userEntityFields = Entity.getFields userEntity
-         in concatMap
-              (validateEntityHasField "app.auth.userEntity" userEntityFields)
-              [ ("username", Entity.Field.FieldTypeScalar Entity.Field.String, "String"),
-                ("password", Entity.Field.FieldTypeScalar Entity.Field.String, "String")
-              ]
+      else validationErrors
+    where
+      validationErrors = concat [usernameValidationErrors, passwordValidationErrors]
+      usernameValidationErrors
+        | not $ null usernameTypeValidationErrors = usernameTypeValidationErrors
+        | otherwise = usernameAttributeValidationErrors
+      passwordValidationErrors =
+        validateEntityHasField
+          userEntityName
+          userEntityFields
+          ("password", Entity.Field.FieldTypeScalar Entity.Field.String, "String")
+      usernameTypeValidationErrors =
+        validateEntityHasField
+          userEntityName
+          userEntityFields
+          ("username", Entity.Field.FieldTypeScalar Entity.Field.String, "String")
+      usernameAttributeValidationErrors
+        | doesFieldHaveAttribute "username" "unique" userEntity == Just True = []
+        | otherwise =
+          [ GenericValidationError $
+              "The field 'username' on Entity referenced by " ++ userEntityName ++ " must have the '@unique' attribute."
```

Comment:
I would go for less technical error message like
```
The username field on the User entity should be marked as unique (add the @unique attribute on the username field).
```

If some new users this super technical message, it might confuse them IMHO

Reply:
I agree, but the rest of the file uses this style so I decided against changing it to save time.

Reply:
Consider changing this message and the others (I see only one other: https://github.com/wasp-lang/wasp/blob/afe90a20e6427e720e9fc39678c38eff5ae8bb35/waspc/src/Wasp/AppSpec/Valid.hs#L245) if you have the time 👍